src/DynamoCoreWpf/Controls/DynamoTextBox.cs

```diff
@@ -344,6 +344,38 @@ public Side AttachmentSide
             get { return ((Side)GetValue(AttachmentSideProperty)); }
             set { SetValue(AttachmentSideProperty, value); }
         }
+
+        private UIElement _target;
+
+        protected override void OnOpened(RoutedEventArgs e)
+        {
+            base.OnOpened(e);
+
+            _target = PlacementTarget;
+            if (_target != null)
+            {
+                _target.IsVisibleChanged += Target_IsVisibleChanged;
+            }
+        }
+
+        protected override void OnClosed(RoutedEventArgs e)
+        {
+            base.OnClosed(e);
+
+            if (_target != null)
+            {
+                _target.IsVisibleChanged -= Target_IsVisibleChanged;
+                _target = null;
+            }
+        }
+
+        private void Target_IsVisibleChanged(object sender, DependencyPropertyChangedEventArgs e)
+        {
+            if (_target != null && !_target.IsVisible)
+            {
+                IsOpen = false;
```

Comment:
Maybe do the unsubscribe here as well and set target to null ?

Reply:
If we unsusbscribe here, the popup will no longer resubscribe after the first disappearance, since tooltips are reused.